---
title: "MAF Tutorial"
author: Nicole Black
date: 08/03/22
---

*/ IMPORTANT: You must query, download, and prepare the RNAseq data before Monday's meeting (see directions in part 8 of this notebook). Allot 15-20 minutes for the download depending on your laptop's processing power. If you encounter any issues, you must slack Nicole/TAs before Sunday night so we can troubleshoot prior to class time. /*

###### (1) Set Up

Set your working environment to analysis_data:

```{r setup}
    knitr::opts_knit$set(root.dir = normalizePath("C:/Users/ac361/Desktop/QBIO_490/qbio_490_avinash/analysis_data/")) 
```

Install and load the ggplot2 package:

```{r}
library(ggplot2)
```

Load in the maftools and TCGAbiolinks package:

```{r}
# write code here
library(maftools)
library(TCGAbiolinks)
```

Load in your saved MAF data by reading in your clinical data, re-preparing maf_query, and then calling read.maf(). See Intro_to_MAF.Rmd if you need help.

```{r}
# write code here
clinical <- read.csv("C:/Users/ac361/Desktop/QBIO_490/qbio_490_avinash/analysis_data/brca_clinical_data.csv")



maf_query <- GDCquery(
    project = "TCGA-BRCA", 
    data.category = "Simple Nucleotide Variation", 
    access = "open", # we only have access to somatic mutations which are open access
    data.type = "Masked Somatic Mutation", 
    workflow.type = "Aliquot Ensemble Somatic Variant Merging and Masking"
)

maf <- GDCprepare(maf_query) # as long as it runs, ignore any errors

maf_object <- read.maf(maf = maf, 
                       clinicalData = clinical,
                       isTCGA = TRUE)
```

###### (2) Oncoplots

We are going to start off by making our first MAF figure using the oncoplot() function. The each column on an oncoplot represents an individual patient, and the rows represent the genes of interest. The different colors represent the different types of mutations where grey represents no mutation in that gene for that patient.

Generate an oncoplot by filling in the maf argument below. Fill in the top argument with various integer arguments (e.g. 5, 10, 20, etc.) to see the genes that are mutated in the most patients. Save your plot to your week6_maf folder.

```{r}
oncoplot(maf = maf_object,
         top = 20) 

ggsave("C:/Users/ac361/Desktop/QBIO_490/qbio_490_avinash/oncoplot2_maf_tutorial.png")
```

Pick one of the top 10 mutated genes and Google it! What is the function of this gene? Remember, at a basic level, cancer is uncontrolled cell growth. Does your gene relate to cell growth and proliferation (division)? If so, how?

DMD 
is a tumor suppressor in human cancers with myogenic programs, providing instructions for making dystrophin. Yes, mutations to DMD prevent an effective balance of short and long dystrophin protein. This ratio has been shown to be clinically significant to tumorigenesis.

You can also create an oncoplot for specific genes of your choosing by passing a vector into oncoplot genes argument. Pick 5 genes to plot. Hint: the genes argument accepts a vector of strings.

```{r}
oncoplot(maf = maf_object,
         genes = c("TP53", "PIK3CA", "TTN", "CDH1", "GATA3"))
```

What other questions might you ask based on the oncoplots created above (think about types of mutations, specific genes, specific pathways, mutual exclusivity of mutations, etc.)? Brainstorm two research questions:

1. How do rates of each type of mutation vary relative to the overall set of genes?
2. Do the mutations work in combination to accumulate the risk of cancer?

###### (3) Adding Clinical Annotations

We can further customize our oncoplots by adding clinical data in the form of an annotation. Let's practice this by adding age_category to our oncoplot.

```{r}
oncoplot(maf = maf_object,
         top = 10,
         clinicalFeatures = "age_category")
```

Do you see any clustering between old and young patients? Yes, there appears to be some clustering for young patients.

###### (4) subsetMaf + CoOncoplots

Another way to compare between patient populations is to use a coOncoplot. Before we create our coOncoplot, we first need to subset our maf_object dataframe into two separate data frames, one for old patients and one for young patients.

We can do this with subsetMaf(). The subsetMaf(maf, tsb) function needs two arguments in this case: maf is the original MAF object, and tsb is a vector of patient IDs (barcodes) to select.

Store the young patient barcodes in a vector called young_patient_barcodes. Hint: you can use boolean masking for this!

```{r}
young_patient_barcodes <- maf_object@clinical.data$Tumor_Sample_Barcode[maf_object@clinical.data$age_category == "young"]
```

Use the young_patient_barcode vector in the tsb argument of subsetMaf.

```{r}
young_maf <- subsetMaf(maf = maf_object,
                       tsb = young_patient_barcodes)
```

Create an old_patient_barcode vector and use that to subset just the old patient data.

```{r}
old_patient_barcodes <- maf_object@clinical.data$Tumor_Sample_Barcode[maf_object@clinical.data$age_category == "old"]
  
old_maf <- subsetMaf(maf = maf_object, tsb = old_patient_barcodes)
```

Use coOncoplot() to draw two oncoplots side by side, comparing young vs. old patients. m1 is the first MAF object to plot, and m2 is the second. Be sure to provide descriptive names in the m1Name and m2Name arguments! Again, save the plot using the ggsave() function.

```{r}
coOncoplot(m1 = young_maf, 
           m2 = old_maf, 
           m1Name = "Age < Fifty Years", 
           m2Name = "Age >= Fifty Years")

jpeg("C:/Users/ac361/Desktop/QBIO_490/qbio_490_avinash/analysis_data/oncoplot2_maf_tutorial.jpg")

dev.off()

```

Are there any genes that seem to be mutated at different rates between the two populations? If so, pick one. Can you think of any reasons why that gene might be mutated more in younger/older patients (Google it!)

Chosen Gene: CDH1
More highly mutated in young or old population: Old population
Hypothesized Reason: Perhaps mutations in this gene accumulate over time, leading to higher rates of expression among older patients.

###### (5) Lollipop Plots

Lollipop plots display the location of each mutation along the geneâ€™s DNA sequence. The x-axis of a lollipop plot is the amino acid sequence that makes up the gene. Individual mutations are annotated by a ball-and-stick, which looks like a lollipop, hence the name. The colors represents the type of mutation at a given point on the gene (similar to colors of the oncoplots), and the visualization in the middle of the plot is the gene itself differentiated between its different domain/motif sections.

Let's make a lollipop plot for TP53.

```{r}
lollipopPlot(maf = maf_object,
             gene = "TP53")

ggsave("C:/Users/ac361/Desktop/QBIO_490/qbio_490_avinash/analysis_data/week6_maf/lollipop_maf_tutorial.png")

dev.off()
```

Does TP53 have any domains? If so, list them and provide a brief description of their functions
1) transcriptional activation domain: 	
The binding of the p53 transactivation domain by regulatory proteins regulates p53 transcription activation
2) sequence-specific DNA-binding domain: exerts antiproliferation effects through iits ability to function as a sequence-soecific DNA-binding transcription factor
3) tetramerization domain: permits the oligomerization of the protein

What is the mutation rate of TP53 among all patients in our population? 34.23%

Approximately what number amino acid is the most mutated, and how many mutations are at this location?
~number 180, 20 mutations

###### (6) Co-Lollipop Plots

Similarly to how we created co-Oncoplots to compare the oncoplot data between two different populations, we can also create co-lollipop plots to do the same.

Recreate your young_maf and old_maf objects.

```{r}
# write code here
young_maf <- subsetMaf(maf = maf_object,
                       tsb = young_patient_barcodes)
old_maf <- subsetMaf(maf = maf_object,
                       tsb = old_patient_barcodes)
```

Now fill in the co-lollipop plot function below. Be sure to add descriptive names for each population.

```{r}
lollipopPlot2(m1 = young_maf, 
              m2 = old_maf, 
              m1_name = "Age < 50 Years Old",
              m2_name = "Age >= 50 Years Old",
              gene = "TP53") ## pick any gene of your choosing to fill in here
```

Now, interpret your plot. For example, answer the following questions:

Does it look like the gene is more commonly mutated in young or old patients? 
Slightly more in young patients
Are there more mutations in a specific region of the protein? Why might that be?
The P53 region has by far the most mutations, which could indicate that this gene has an oversized effect when directly mutated relative to the other regions.
What kinds of mutations are the most common?
Missense mutations
Are there any other interesting observations?
A number of mutations are associated with unnamed regions, but it is unclear what these regions represent.

###### (7) Survival Analysis (mafSurvival)

Remember back to Kaplan-Meier survival plots (from the clinical data tutorial). These plots allow us to visualize survival probabilities stratified by different clinical factors (such as age, sex, stage, etc.)

We can also visualize survival probabilities stratified by mutational status. To do this, we use mafSurvival. This is very similar to our traditional KM plots, you may want to open up the clinical data tutorial to follow along there as well.

```{r}
maf_object@clinical.data$days_to_last_followup = ifelse(!is.na(maf_object@clinical.data$days_to_death), maf_object@clinical.data$days_to_death, maf_object@clinical.data$days_to_last_followup)


maf_object@clinical.data$Overall_Survival_Status <- ifelse(maf_object@clinical.data$vital_status == "Dead", TRUE, FALSE)



mafSurvival(maf = maf_object,
            genes = "TP53", ## pick a gene of your choosing
            time = "days_to_last_followup", ## name of the column in maf_object@clinical.data containing survival time, you may need to recreate this... if so, check the clinical data tutorial for instructions
            Status = "Overall_Survival_Status", ## name of the column that contains a boolean value for death events, you may need to recreate this... if so, check the clinical data tutorial for instructions
            isTCGA = TRUE)
```

Provide a brief description of your chosen gene and it's posited relationship to survival.

What can you conclude from your plot?

Is this what you would expect according to the literature? Why or why not?

###### (8) Looking Ahead - VERY IMPORTANT!!

Query, download, and prepare the RNA Transcriptomics data in to your R work space. Confirm that your working directory is analysis_data before you run these lines.

Note that this is super slow!! Don't worry, just wait it out.

```{r}
rna_query <- GDCquery(project = "TCGA-BRCA",
                      data.category = "Transcriptome Profiling",
                      data.type = "Gene Expression Quantification",
                      workflow.type = "STAR - Counts")

GDCdownload(rna_query)

rna_se <- GDCprepare(rna_query)
```
